### PR TITLE
Update docs for 0.4.0 release

### PR DESCRIPTION
**Pull request by foxish**
ptal @erikerlandson @ash211 @mccheah @liyinan926 @ifilonenko

### CODE REVIEW

File: src/jekyll/running-on-kubernetes.md [hunk 1]
```diff
@@ -225,7 +237,9 @@ Note that this resource staging server is only required for submitting local dep
 dependencies are all hosted in remote locations like HDFS or http servers, they may be referred to by their appropriate
 remote URIs. Also, application dependencies can be pre-mounted into custom-built Docker images. Those dependencies
 can be added to the classpath by referencing them with `local://` URIs and/or setting the `SPARK_EXTRA_CLASSPATH`
-environment variable in your Dockerfiles.
+environment variable in your Dockerfiles. For any remote dependencies that aren't baked into the driver and executor
+docker images, whether they are supplied via http, or hdfs, or the resource staging server,  the
```
Comment (liyinan926) [inline src/jekyll/running-on-kubernetes.md]: any remote dependencies (not using the `local://` scheme) ...

**Comment by liyinan926** [comment]
LGTM with a minor comment.

File: src/jekyll/running-on-kubernetes.md (hunk 1 above)
Comment (foxish) [reply src/jekyll/running-on-kubernetes.md]: Done

**Comment by erikerlandson** [comment]
so the image tags for python got a `spark-` prefix now?  I'm not seeing that up on the docker registry, the 0.4.0 images are still `driver-py` and `executor-py`

**Comment by foxish** [comment]
Yes, they were standing out - the script that pushes images now pushes them up as spark-driver-py and spark-executor-py, in line with the other images.

**Comment by erikerlandson** [comment]
I'm not seeing `spark-driver-py` or `spark-executor-py` up on docker, its still the old names

**Comment by liyinan926** [comment]
No, I just realized that we now have that script but I didn't use the script. Will repush driver-py and executor-py.

**Comment by liyinan926** [comment]
Done.

**Comment by erikerlandson** [comment]
LGTM

**Pull request merged**